### PR TITLE
Deprecate `is_initialized` in favor of `initialized` and `finalized`

### PR DESCRIPTION
**Pull request by dalg24**
*My* proposed resolution for #5208

### CODE REVIEW

File: core/src/impl/Kokkos_Core.cpp [hunk 1]
```diff
@@ -666,7 +667,7 @@ void finalize_internal() {
 
   Kokkos::Impl::ExecSpaceManager::get_instance().finalize_spaces();
 
-  g_is_initialized = false;
+  g_finalized      = false;
```
Comment (PhilMiller) [inline core/src/impl/Kokkos_Core.cpp:670]: Should be true here?

**Review by crtrott — APPROVED** [review]
Please document the deprecation.

**Review by nliber — APPROVED** [review]

**Review by crtrott — CHANGES_REQUESTED** [review]
Ok putting a hold on it lets talk about the options today, I thought I missed that the team had agreed on something, and was just a bit done with arguing :-)

**Comment by dalg24** [comment]
Dev team decided to go for option 1

**Pull request closed without merging**